### DISCUSSION TITLE
How should I handle the form state during waiting the respone?

### DISCUSSION THREAD

**Discussion by pkino** [react-hook-form/react-hook-form#3358] [Q&A]
I know formState `isSubmitting` is useful to know about the state of a form.
However, `isSubmitting` returns false after submitted but before arriving the response.
I want to get the status from the beginning of submitting to the arrival of the response.
Currently, I create the variables of useState and set true from the beginning of submitting to the arrival of the response.
Are there any good ways other than that?

**Comment by bluebill1049** [comment]
> However, isSubmitting returns false after submitted but before arriving the response.

What do you mean can you share that in a CSB?

**Reply by pkino** [reply]
I mean I want to disable the submit button until the request is completed (so, I use the state `isLoading`).
https://codesandbox.io/s/suspicious-field-3jb0c

**Reply by bluebill1049** [reply]
```
const onSubmit = async (data: Inputs): void => {
    setIsLoading(true);
    await axios({
      url: "/api/login",
      method: "post",
      data: data
    })
      .then(() => {})
      .catch((error) => {
        setIsLoading(false);
      });
  };
```

`isSubmitting` should then worked.

**Reply by pkino** [reply]
`isSubmitting` returns false before going this `then`.
So, we can push the submit button again, before going the next page.

```
const onSubmit = async (data: Inputs): void => {
    setIsLoading(true);
    await axios({
      url: "/api/login",
      method: "post",
      data: data
    })
      .then(() => {})
      .catch((error) => {
        setIsLoading(false);
      });
  };
```

**Reply by bluebill1049** [reply]
are you readng `isSubmitting` at render?

**Reply by pkino** [reply]
Yes, I am reading it like in the CSB.
If you have a valid sample, could you share that?

**Reply by bluebill1049** [reply]
https://github.com/react-hook-form/react-hook-form/tree/master/examples

there is an example there.

**Reply by pkino** [reply]
I checked that example, my implementation is the same as the example.
I want to realize implementation like this.
I want to use `isSubmitting` as same as like ` isLoading`.
https://codesandbox.io/s/keen-boyd-ym6iz

**Reply by pkino** [reply]
Didn't `isSbumitting` intend the usage above?
